haskell/haddock.bzl

```diff
@@ -232,30 +238,36 @@ def _haskell_doc_rule_impl(ctx):
     if ctx.attr.index_transitive_deps:
         # Include all packages in the unified index.
         for package_id in html_dict_copied:
-            args.add("--read-interface=../{0},{1}".format(
-                package_id,
-                haddock_dict[package_id].path,
-            ))
+            for interface in haddock_dict[package_id]:
+                args.add("--read-interface=../{0},{1}".format(
+                    package_id,
+                    interface.path,
+                ))
     else:
         # Include only direct dependencies.
         for dep in ctx.attr.deps:
             if HaddockInfo in dep:
                 package_id = dep[HaddockInfo].package_id
-                args.add("--read-interface=../{0},{1}".format(
-                    package_id,
-                    haddock_dict[package_id].path,
-                ))
+                for interface in haddock_dict[package_id]:
+                    args.add("--read-interface=../{0},{1}".format(
+                        package_id,
+                        interface.path,
+                    ))
 
     args.add_all(all_caches, map_each = _dirname, format_each = "--optghc=-package-db=%s")
     locale_archive_depset = (
         depset([hs.toolchain.locale_archive]) if hs.toolchain.locale_archive != None else depset()
     )
 
+    haddocks = []
```

Comment:
You could also use
```
haddocks = depset(transitive = [depset(i) for i in transitive_haddocks.values()])
```
If `transitive_haddocks` contained `depset`s then you could also drop the list comprehension.

Same in `_haskell_doc_aspect_impl`.

Reply:
Thank you!